core/src/main/scala/scalaz/BindRec.scala

```diff
@@ -47,9 +49,40 @@ trait BindRec[F[_]] extends Bind[F] { self =>
   val bindRecSyntax = new scalaz.syntax.BindRecSyntax[F] { def F = BindRec.this }
 }
 
+trait MonadRecSafely[M[_], A] {
```

Comment:
https://travis-ci.org/scalaz/scalaz/jobs/313239337#L811

[checkGenTypeClasses](https://github.com/scalaz/scalaz/blob/68ee32f5bbc25126935d98e62cbfb81144fe7f5b/project/build.scala#L180) task failed. please separate to a new file 🙏